fish-rust/src/compat.rs

```diff
@@ -33,12 +33,67 @@ pub static _PATH_BSHELL: AtomicPtr<i8> = AtomicPtr::new(std::ptr::null_mut());
 #[allow(non_snake_case)]
 pub static _PC_CASE_SENSITIVE: Lazy<c_int> = Lazy::new(|| unsafe { C_PC_CASE_SENSITIVE() });
 
+macro_rules! CVAR {
+    ($cfn:ident, $cvar:ident, $type:ident) => {
+        #[allow(non_snake_case)]
+        pub fn $cvar() -> $type {
+            unsafe { $cfn() }
+        }
+    };
+}
+
+CVAR!(C_RLIMIT_SBSIZE, RLIMIT_SBSIZE, i32);
```

Comment:
> All of the RLIMIT variables are present in libc but AFAIK there is no way to check if each one of them exists like there is in C/C++ using #ifdef

Right. The limits are defined as enum variants like

        #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
        #[cfg_attr(docsrs, doc(cfg(all())))]
        /// The maximum size (in bytes) of socket buffer usage for this user.
        RLIMIT_SBSIZE,

We don't want to use `#[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]` because that requires extra work whenever OSs change.

We could outsource the checks to a compiler invocation in `build.rs`, similar to `compiles("fish-rust/src/cfg/w_exitcode.cpp")`.
I'm not sure how to correctly invoke `rustc`. We could use a C compiler or preprocesor too..

Probably too complicated for now. It's fine to leave it. Maybe there is an established pattern for this kind of problem.